File: VB/E494VB/WinSolution.Module.Win/MyDynamicListViewVariantFilterController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.ViewVariantsModule
Imports DevExpress.Data.Filtering

Namespace WinSolution.Module.Win
	Partial Public Class MyDynamicListViewVariantFilterController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
		End Sub
		Protected Overrides Overloads Sub OnFrameAssigned()
			MyBase.OnFrameAssigned()
			AddHandler Frame.GetController(Of ChangeVariantController)().ChangeVariantAction.Executed, AddressOf ChangeVariantAction_Executed
		End Sub
		Private criteria As CriteriaOperator = Nothing
		Private Sub ChangeVariantAction_Executed(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.Actions.ActionBaseEventArgs)
			If Frame.View.Id = "DomainObject1_ListView" Then
				criteria = CriteriaOperator.Parse("FirstName LIKE 'A%'")
			Else
				If Frame.View.Id = "DomainObject1_ListView_Variant1" Then
					criteria = CriteriaOperator.Parse("FirstName LIKE 'B%'")
				End If
			End If
			CType(Frame.View, ListView).CollectionSource.Criteria("MyFilter") = criteria
		End Sub
	End Class
End Namespace

File: CS/WinSolution.Module.Win/MyDynamicListViewVariantFilterController.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.ViewVariantsModule;
using DevExpress.Data.Filtering;

namespace WinSolution.Module.Win {
    public partial class MyDynamicListViewVariantFilterController : ViewController {
        public MyDynamicListViewVariantFilterController() {
            InitializeComponent();
            RegisterActions(components);
        }
        protected override void OnFrameAssigned() {
            base.OnFrameAssigned();
            Frame.GetController<ChangeVariantController>().ChangeVariantAction.Executed += new System.EventHandler<DevExpress.ExpressApp.Actions.ActionBaseEventArgs>(ChangeVariantAction_Executed);
        }
        CriteriaOperator criteria = null;
        void ChangeVariantAction_Executed(object sender, DevExpress.ExpressApp.Actions.ActionBaseEventArgs e) {
            if (Frame.View.Id == "DomainObject1_ListView") {
                criteria = CriteriaOperator.Parse("FirstName LIKE 'A%'");
            }
            else
                if (Frame.View.Id == "DomainObject1_ListView_Variant1") {
                    criteria = CriteriaOperator.Parse("FirstName LIKE 'B%'");
                }
            ((ListView)Frame.View).CollectionSource.Criteria["MyFilter"] = criteria;
        }
    }
}
